apps/web/stores/AnswerStore.ts

```diff
@@ -22,6 +19,25 @@ export const DEFAULT_ANSWER_VALUE_MULTI_CHOICE = null;
 export const DEFAULT_ANSWER_VALUE_NUMBER_FLOAT = NaN;
 export const DEFAULT_ANSWER_VALUE_MULTI_CHOICE_MULTI: string[] = [];
 
+export const answerValuesAreNotEqual = (
+  newAnswerValue: AnswerTypes,
+  currentAnswerValue: AnswerTypes,
+): boolean => {
+  if (isArray(newAnswerValue) && isArray(currentAnswerValue)) {
+    return !newAnswerValue.every((value) => currentAnswerValue.includes(value));
+  } else if (isString(newAnswerValue) && isString(currentAnswerValue)) {
+    return newAnswerValue !== currentAnswerValue;
+  } else if (isNumber(newAnswerValue) && isNumber(currentAnswerValue)) {
+    return newAnswerValue !== currentAnswerValue;
+  } else if (isObject(newAnswerValue) && isObject(currentAnswerValue)) {
+    return !Object.entries(newAnswerValue).every(
+      ([key, value]) => currentAnswerValue[key] === value,
+    );
+  }
+
+  throw new Error("Invalid types");
```

Comment:
Is this being handled up the chain with a try catch?

Reply:
Yeah, this is only called from `handleVariableItem` which is only called from `handleForwardNavigation` which is only called recursively in `handleVariableItem` or itself OR in the try catch in `Walkthrough.tsx`